statsd/statsd.go

```diff
@@ -331,6 +335,11 @@ func (c *Client) watch() {
 	for {
 		select {
 		case <-ticker.C:
+			// c.debugDeadlock is always nil except for the TestCloseDeadlock test.
+			if c.debugDeadlock != nil {
+				<-c.debugDeadlock
+			}
+
```

Comment:
Could we instead overwrite the `NewTicker` function ? Like we do [here](https://github.com/DataDog/datadog-agent/blob/master/pkg/metadata/scheduler.go#L25).

That way the code logic is the same when running tests or not and we can control when the ticker triggers.

Reply:
@hush-hush - do you think you can just take this PR over in a separate branch and make the changes yourself? I am a datadog customer and rely on datadog-go to integrate my company's product into the datadog stack. This bug is preventing me from upgrading our dependency and taking advantage of new features.

Reply:
After looking more into this I think it's fine as it is :+1: